src/index.ts

```diff
@@ -111,3 +129,27 @@ export async function generatePdf(
   const mergedPdfBuffer = mergePdfBuffers(generatedPdfBuffers);
   fs.writeFileSync(`${filename}`, mergedPdfBuffer);
 }
+
+export async function generatePdfFromBuildSrources(
+  buildDirPath: string,
+  firstDocPath: string,
+  baseUrl: string,
+  filename: string = "docusaurus.pdf"
+): Promise<void> {
+  let app = express();
+
+  baseUrl = getPathSegment(baseUrl, false);
+  firstDocPath = getPathSegment(firstDocPath);
+
+  let httpServer = await app.listen();
+  let address = httpServer.address();
+  if (!address || !isAddressInfo(address)) {
+    httpServer.close();
+    throw new Error("Something went wrong spinning up the express webserver.");
+  }
+
+  app.use(baseUrl, express.static(buildDirPath));
```

Comment:
I have a question and hopefully someone (you?) can help me understanding:

_The function works as you can see it here, but ..._

Serving the static files failes when I move the `app.use(baseUrl, express.static(...))` command ahead of `app.listen()`. For example:

```
let app = express();
app.use(baseUrl, express.static(buildDirPath));

baseUrl = getPathSegment(baseUrl, false);
firstDocPath = getPathSegment(firstDocPath);

let httpServer = await app.listen();
...
```

I did not understand why this is happening. Normally when I work with `express` I do such `app.use()` calls always before calling the `.listen()` function.

Do you have any idea what goes wrong if I change this to the _normal_ order?